src/support/json.h

```diff
@@ -257,12 +258,22 @@ struct Value {
     while (*curr && is_json_space(*curr))                                      \
       curr++;                                                                  \
   }
+#define skip_escaped_characters(ptr)                                           \
+  while (*ptr && *ptr != '"') {                                                \
+    if (*ptr == '\\' && *(ptr + 1)) {                                          \
+      ptr++;                                                                   \
+    }                                                                          \
+    ptr++;                                                                     \
+  }
     skip();
     if (*curr == '"') {
       // String
       curr++;
-      char* close = strchr(curr, '"');
-      assert(close);
+      char* close = curr;
+      skip_escaped_characters(close);
+      if (!(*close == '"')) {
+        wasm::Fatal() << "Assertion failed (close == '\"') ";
```

Comment:
```suggestion
        wasm::Fatal() << "missing end quote";
```
(because "assertion" suggests it might be a debug-only assertion)